mender-convert-modify

```diff
@@ -233,78 +233,73 @@ EOF
-            mender_client_deb_name="${DEB_NAME}"
-            ;;
-        *)
-            # For 4.x, master and latest(*) install the new packages
-            log_info "Installing Mender Auth and Mender Update version ${mender_client_deb_version}"
-            deb_get_and_install_package mender-client4 mender-auth "${mender_client_deb_version}"
-            deb_get_and_install_package mender-client4 mender-update "${mender_client_deb_version}"
-            mender_client_deb_name="${DEB_NAME}"
-
-            log_info "Installing Mender Flash version ${MENDER_FLASH_VERSION}"
-            deb_get_and_install_package mender-flash mender-flash "${MENDER_FLASH_VERSION}"
-
-            if [ "$MENDER_SETUP_INSTALL" = "auto" ]; then
-                MENDER_SETUP_INSTALL=y
-            fi
-
-            if [ "$MENDER_SNAPSHOT_INSTALL" = "auto" ]; then
-                MENDER_SNAPSHOT_INSTALL=y
-            fi
-            ;;
-    esac
+    mender_client_deb_version="$MENDER_CLIENT_VERSION"
+
+    log_info "Installing Mender Client version ${mender_client_deb_version}"
+    pkgs_to_install["mender-client4"]="$mender_client_deb_version"
 
-    if [ "$MENDER_SETUP_INSTALL" = "y" ]; then
+    log_info "Installing Mender Flash version ${MENDER_FLASH_VERSION}"
+    pkgs_to_install["mender-flash"]="${MENDER_FLASH_VERSION}"
+
+    if [ "$MENDER_SETUP_INSTALL" = "auto" ] || [ "$MENDER_SETUP_INSTALL" = "y" ]; then
         log_info "Installing Mender Setup version ${MENDER_SETUP_VERSION}"
-        deb_get_and_install_package mender-setup mender-setup "${MENDER_SETUP_VERSION}"
+        pkgs_to_install["mender-setup"]="${MENDER_SETUP_VERSION}"
     fi
 
-    if [ "$MENDER_SNAPSHOT_INSTALL" = "y" ]; then
+    if [ "$MENDER_SNAPSHOT_INSTALL" = "auto" ] || [ "$MENDER_SNAPSHOT_INSTALL" = "y" ]; then
         log_info "Installing Mender Snapshot version ${MENDER_SNAPSHOT_VERSION}"
-        deb_get_and_install_package mender-snapshot mender-snapshot "${MENDER_SNAPSHOT_VERSION}"
+        pkgs_to_install["mender-snapshot"]="${MENDER_SNAPSHOT_VERSION}"
     fi
-
-    # Save installed client version for tests in Yocto variable format
-    testscfg_add "PREFERRED_VERSION:mender-client" "$(echo ${mender_client_deb_name} | sed -r 's/.*_([0-9]+\.[0-9]+\.[0-9]+).*/\1/')"
 fi
 
 if [ "${MENDER_ADDON_CONNECT_INSTALL}" = "y" ]; then
     log_info "Installing Mender Connect addon"
-    deb_get_and_install_package mender-connect mender-connect "${MENDER_ADDON_CONNECT_VERSION}"
+    pkgs_to_install["mender-connect"]="${MENDER_ADDON_CONNECT_VERSION}"
 fi
 
 if [ "${MENDER_ADDON_CONFIGURE_INSTALL}" = "y" ]; then
     log_info "Installing Mender Configure addon"
-    deb_get_and_install_package mender-configure mender-configure "${MENDER_ADDON_CONFIGURE_VERSION}" "true"
+    pkgs_to_install["mender-configure"]="${MENDER_ADDON_CONFIGURE_VERSION}"
+fi
+
+if [[ ${#pkgs_to_install[@]} -gt 0 ]]; then
+    deb_ensure_repo_enabled
+    if echo ${pkgs_to_install[@]} | grep -qF "master"; then
```

Comment:
```suggestion
    if [[ ${pkgs_to_install[@]} =~ master ]]; then
```
probably cheaper. if master is at the end of beginning of the string you can probably use %% or ##  to check the prefix/suffix.

Reply:
It can be anywhere.